run_project_tests.py

```diff
@@ -1410,6 +1424,53 @@ def tqdm_print(*args: mlog.TV_Loggable, sep: str = ' ') -> None:
     ET.ElementTree(element=junit_root).write(xmlname, xml_declaration=True, encoding='UTF-8')
     return passing_tests, failing_tests, skipped_tests
 
+def check_file(file: Path):
+    lines = file.read_bytes().split(b'\n')
+    tabdetector = re.compile(br' *\t')
+    for i, line in enumerate(lines):
+        if re.match(tabdetector, line):
+            raise SystemExit("File {} contains a tab indent on line {:d}. Only spaces are permitted.".format(file, i + 1))
+        if line.endswith(b'\r'):
+            raise SystemExit("File {} contains DOS line ending on line {:d}. Only unix-style line endings are permitted.".format(file, i + 1))
+
+def check_format():
+    check_suffixes = {'.c',
+                      '.cpp',
+                      '.cxx',
+                      '.cc',
+                      '.rs',
+                      '.f90',
+                      '.vala',
+                      '.d',
+                      '.s',
+                      '.m',
+                      '.mm',
+                      '.asm',
+                      '.java',
+                      '.txt',
+                      '.py',
+                      '.swift',
+                      '.build',
+                      '.md',
+                      }
+    skip_dirs = {
+        '.dub',                         # external deps are here
+        '.pytest_cache',
+        'meson-logs', 'meson-private',
+        'work area',
+        '.eggs', '_cache',              # e.g. .mypy_cache
+        'venv',                         # virtualenvs have DOS line endings
+    }
+    for (root, _, filenames) in os.walk('.'):
+        if any([x in root for x in skip_dirs]):
+            continue
+        for fname in filenames:
+            file = Path(fname)
+            if file.suffix.lower() in check_suffixes:
+                if file.name in ('sitemap.txt', 'meson-test-run.txt'):
+                    continue
+                check_file(root / file)
+
```

Comment:
This, and a few other bits, seems independent?

Reply:
Mistakes from rebasing old changesets; thanks for your patience!